File: Frontend/friend-nest/src/app/models/response/PostTo.ts
import { Post } from '../Post';
import { UserInteractions } from './UserInteractionsToPost';
import { User } from '../User';

export interface PostTo {
  quotedPost?: PostTo;
  reposter?: User;
  post: Post;
  isReposted?: boolean;
  isQuoted?: boolean;
  userInteractions: UserInteractions;
}

File: Backend/friendnest/src/main/java/pl/polsl/friendnest/model/PostTo.java
package pl.polsl.friendnest.model;

import lombok.*;
import pl.polsl.friendnest.model.response.UserInteractions;

@NoArgsConstructor
@Getter
@Setter
@Builder
@AllArgsConstructor
public class PostTo {
    private Post post;
    private Boolean isReposted;
    private Boolean isQuoted;
    private User reposter;
    private PostTo quotedPost;
    private UserInteractions userInteractions;
}
